src/main/js/field/ProviderDataSelectField.js

```diff
@@ -0,0 +1,162 @@
+        const resolvedEndpoint = endpoint.replace('{provider}', providerName);
+        const requestUrl = `/alert/api${resolvedEndpoint}`;
+        return fetch(requestUrl, {
+            credentials: 'same-origin'
+        })
+            .then((response) => {
+                response.json()
+                    .then((json) => {
+                        if (!response.ok) {
+                            verifyLoginByStatus(response.status);
+                            this.providerDataError(true, 'There was a problem with the request');
+                        } else {
+                            const providerData = json.map(item => {
+                                const dataValue = item.value;
+                                return { icon: null, key: dataValue, label: dataValue, value: dataValue };
+                            });
+                            this.providerDataFetched(providerData);
+                        }
+                    });
+            })
+            .catch((error) => {
+                console.log(`Unable to connect to Server: ${error}`);
+                this.providerDataError(true, error);
+            });
+    }
+
+    render() {
+        const {
+            id, inputClass, searchable, placeholder, value, removeSelected, multiSelect, components, selectSpacingClass, readOnly, onChange
+        } = this.props;
+
+        const selectClasses = `${selectSpacingClass} d-inline-flex p-2`;
+
+        const handleChange = (selectedOptions) => {
+            const optionValue = selectedOptions ? selectedOptions.value : null;
+            const parsedArray = (Array.isArray(selectedOptions) && selectedOptions.length > 0) ? selectedOptions.map(mappedOption => mappedOption.value) : optionValue;
+            onChange({
+                target: {
+                    name: id,
+                    value: parsedArray
+                }
+            });
+        };
+
+        const { providerData } = this.state;
+        const options = providerData ? providerData : [];
+        const selectValue = options.filter(option => value.includes(option.value));
+
+        const field = (<div className={selectClasses}>
+            <Select
+                id={id}
+                className={inputClass}
+                onChange={handleChange}
+                isSearchable={searchable}
+                removeSelected={removeSelected}
+                options={options}
+                placeholder={placeholder}
+                value={selectValue}
+                isMulti={multiSelect}
+                closeMenuOnSelect={!multiSelect}
+                components={components}
+                isDisabled={readOnly}
+                noOptionsMessage={this.noOptionsMessage}
+            />
+        </div>);
+        return (
+            <LabeledField field={field} {...this.props} />
+        );
+    }
+}
+
+ProviderDataSelectField.propTypes = {
+    id: PropTypes.string,
+    providerDataEndpoint: PropTypes.string,
```

Comment:
you could make this prop required as it seems vital to this components functionality.